index.d.ts

```diff
@@ -0,0 +1,15 @@
+import through2 = require("through2");
+export = connect
+type TParser<JSON> = {
+  [Property in keyof JSON]: (variable: any) => JSON[Property]
+}
+type TOptions = {
+  uri?: string
+  parse?: () => TParser<JSON>
+}
+
```

Comment:
```suggestion
type JSONValue =
    | string
    | number
    | boolean
    | { [x: string]: JSONValue }
    | Array<JSONValue>
type TParser = (input: string) => JSONValue
type TOptions = {
    uri?: string
    parse?: TParser
}
```

Reply:
I think your types are a bit off here. The parser option is essentially just a function that takes a string and returns a JSON serializable value.